pkg/bootstrap/config.go

```diff
@@ -115,7 +116,10 @@ func (cfg Config) toTemplateParams() (map[string]any, error) {
 		option.XdsType(xdsType),
 		option.MetadataDiscovery(bool(metadataDiscovery)),
 		option.MetricsLocalhostAccessOnly(cfg.Metadata.ProxyConfig.ProxyMetadata),
-		option.DeferredClusterCreation(features.EnableDeferredClusterCreation))
+		option.DeferredClusterCreation(features.EnableDeferredClusterCreation),
+		option.IstioMinorVersion(iVersion.Minor),
```

Comment:
This code doesn't need to deal with version skew. This is the agent, which is always tightly coupled with envoy/the bootstrap file. We only need to handle the skew in the control plane.

This also might have been why you had to move the packages around?

Reply:
Ah good catch - not sure why I thought we needed a different regex in the boostrap file. That'll vastly improve the diff as well 👍🏾